node/network/collator-protocol/src/lib.rs

```diff
@@ -53,13 +53,28 @@ enum Error {
 	RuntimeApi(RuntimeApiError),
 	#[from]
 	UtilError(util::Error),
+	#[from]
+	Prometheus(prometheus::PrometheusError),
 }
 
 type Result<T> = std::result::Result<T, Error>;
 
 enum ProtocolSide {
-	Validator,
-	Collator(CollatorId),
+	Validator(validator_side::Metrics),
+	Collator(CollatorId, collator_side::Metrics),
+}
+
+/// Metrics.
+#[derive(Clone, Default)]
+pub struct Metrics;
+
+// TODO: So how do we tie this to metrics in modules?.
```

Comment:
yeah, so the assumption is that a subsystem has one `Metrics` type, but it doesn't really work for the collator protocol subsystem, we can also use `()` as a placeholder or create an additional enum, but it's not ideal